app/js/components/orderEntry/OrdersTable.jsx

```diff
@@ -9,21 +9,86 @@ import DrugOrderDetails from './DrugOrderDetails';
 import fetchOrders from '../../actions/fetchOrders';
 import { DRUG_ORDER } from './orderTypes';
 import { setSelectedOrder } from '../../actions/orderAction';
+import createLabOrder from '../../actions/createLabOrder';
+import { successToast, errorToast } from '../../utils/toast';
+import toggleUrgency from '../../actions/labOrders/toggleUrgency';
 
 export class OrdersTable extends PureComponent {
+  state = {
+    recentUrgency: '',
+    previousOrderId: '',
+  };
+
   componentDidMount() {
     this.props.dispatch(fetchOrders(null, this.props.patient.uuid));
   }
 
-  componentDidUpdate(prevProps, prevState) {
+  componentDidUpdate(prevProps) {
+    const { previousOrderId, recentUrgency } = this.state;
+    const {
+      status: { added, error },
+      errorMessage,
+      labOrderData,
+    } = this.props.createLabOrderReducer;
     if (this.props.patient.uuid !== prevProps.patient.uuid) {
       this.props.dispatch(fetchOrders(null, this.props.patient.uuid));
     }
+    if (added && labOrderData !== prevProps.createLabOrderReducer.labOrderData) {
+      this.props.dispatch(toggleUrgency(
```

Comment:
I don't think it makes sense to be dispatching an action inside of a component lifecycle, @IAMOTZ coman carry your brother

Reply:
It's bug prone, especially if this implementation should be scaled in some way

Reply:
in your question `so, how and where is asynchronous API calls (fetching of data) done when a page is loaded done??` not when a component is been updating, preferably before initial render but definitely not when it's updating, and handler should be used to trigger updates. you can actually call `setState` in lifecycles which although it's still ill advised but definitely not in `render()`. Yeah it doesn't matter if it's not asynchronous or not, it's just not cool to update state state in the middle of an update, think about it...

Reply:
your argument initially is that actions should not be dispatched in `componentLifeCycle` methods, the question I asked was just to bring to your attention that it is done and very necessary in a `React` project.

I also want to emphasize that I am not calling setState here, I am only dispatching an action that would trigger a `re-render` since the state would change. I only mentioned `setState` as an instance of the fact that a re-render can be triggered after a component is updated if necessary but should be done properly so that an update loop doesn't happen.